sdk/resourcemanager/azure-resourcemanager-monitor/src/test/java/com/azure/resourcemanager/monitor/DiagnosticSettingsTests.java

```diff
@@ -198,6 +198,88 @@ public void canCRUDDiagnosticSettingsForSubscription() {
         }
     }
 
+    @Test
+    public void canCRUDDiagnosticSettingsForStorageAccount() {
+
+        Region region = Region.US_WEST;
+
+        StorageAccount sa = storageManager.storageAccounts()
+                .define(saName)
+                .withRegion(region)
+                .withNewResourceGroup(rgName)
+                .withTag("tag1", "value1")
+                .create();
+
+        // clean all diagnostic settings.
+        List<DiagnosticSetting> dsList = monitorManager.diagnosticSettings().listByResource(sa.id()).stream().collect(Collectors.toList());
+        for (DiagnosticSetting dsd : dsList) {
+            monitorManager.diagnosticSettings().deleteById(dsd.id());
+        }
+
+        EventHubNamespace namespace = eventHubManager.namespaces()
+                .define(ehName)
+                // EventHub should be in the same region as resource
+                .withRegion(region)
+                .withNewResourceGroup(rgName)
+                .withNewManageRule("mngRule1")
+                .withNewSendRule("sndRule1")
+                .create();
+
+        EventHubNamespaceAuthorizationRule evenHubNsRule = namespace.listAuthorizationRules().iterator().next();
+
+        List<DiagnosticSettingsCategory> categories = monitorManager.diagnosticSettings()
+                .listCategoriesByResource(sa.id());
+
+        Assertions.assertNotNull(categories);
+        Assertions.assertFalse(categories.isEmpty());
+
+        DiagnosticSetting setting = monitorManager.diagnosticSettings()
+                .define(dsName)
+                .withResource(sa.id())
+                .withEventHub(evenHubNsRule.id())
+                .withLogsAndMetrics(categories, Duration.ofMinutes(5), 7)
+                .create();
+
+        Assertions.assertTrue(sa.id().equalsIgnoreCase(setting.resourceId()));
+        Assertions.assertNull(setting.storageAccountId());
+        Assertions.assertTrue(evenHubNsRule.id().equalsIgnoreCase(setting.eventHubAuthorizationRuleId()));
+        Assertions.assertNull(setting.eventHubName());
+        Assertions.assertNull(setting.workspaceId());
+        Assertions.assertFalse(setting.logs().isEmpty());
```

Comment:
I failed at this line. Seems we don't have resource log category for storage accounts. 

Do you mind if I change it to diagnostic settings for key vault? Otherwise it'll fail our live tests afterwards...

Reply:
That would be lovely, thanks! I've been having a lot of trouble running the tests locally on my Mac